src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -94,14 +95,23 @@ class ADRController(private val encryption: Encryption,
             "$url&hide_inaccessible_resources=true"
         }
         val response = adr.get(url)
-        return if (response.statusCode != HttpStatus.OK)
+        if (response.statusCode != HttpStatus.OK)
         {
-            response
+            var errorKey = "adrFetchingDatasetsError"
+            if (response.body?.contains("has not yet logged into ADR") == true)
+            {
+                errorKey = "adrFetchingDatasetsNoAccountError"
+            }
```

Comment:
Something like this would work: 

```suggestion
           val errorKey = if (response.body?contains("has not yet logged into ADR)) "adrFetchingDatasetsNoAccountError" else "adrFetchingDatasetsError"         }
```